custom_components/variable/__init__.py

```
@@ -222,75 +251,106 @@ async def _async_process_yaml(hass: HomeAssistant, config: ConfigType) -> bool:
+                _LOGGER.error(
+                    "[YAML] Cannot import %s because that variable ID belongs to a UI-created entry",
+                    var,
+                )
+                continue
 
-            if var not in {
-                entry.data.get(CONF_VARIABLE_ID)
-                for entry in hass.config_entries.async_entries(DOMAIN)
-            }:
+            if not yaml_entries:
                 _LOGGER.warning(f"[YAML] Creating New Sensor Variable: {var}")
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)